dkc/core/rest/folder.py

```diff
@@ -135,18 +135,19 @@ def get_serializer_context(self):
     @transaction.atomic
     def perform_create(self, serializer: serializers.ModelSerializer):
         parent: Folder = serializer.validated_data.get('parent')
```

Comment:
The use of `FullCleanModelSerializer` is causing this to fail.

Read-only fields or non-serialized fields, like `creator`, are now required to be present on the input in order for `serializer.validated_data` to succeed.

Traditionally, `ModelViewSet.perform_create` (which runs after the serializer validates) is the right place to tack on additional mandatory, implicitly determined, data ([the DRF docs encourage this too](https://www.django-rest-framework.org/tutorial/6-viewsets-and-routers/#refactoring-to-use-viewsets)).

---

There is a solution: use `creator = ForeignKey(editable=False, ...)`. This causes `Model.full_clean()` to skip validation of that field (more precisely, [when the field validates itself, it is a no-op](https://github.com/django/django/blob/6822aa5c6c3fbec7c5393a05e990865ba59fe167/django/db/models/fields/__init__.py#L637)); this is the only reason that the `tree` field is able to work.

However, I don't think we want the `creator` to be non-editable (i.e. Django admins should be allowed to change this field, even if end users can't). As we develop other definitely-non-editable fields which require implicit creation, this problem will persist.